opm/simulators/wells/ALQState.cpp

```diff
@@ -38,7 +38,8 @@ double ALQState::get(const std::string& wname) const {
     if (default_iter != this->default_alq_.end())
         return default_iter->second;
 
-    throw std::logic_error("No ALQ value registered for well: " + wname);
+    // injectors does not have a default alq return 0.0
+    return 0.0;
```

Comment:
Could we solve this another way, e.g., by not calling the function if the well is an injector?  Blindly returning zero seems a little presumptuous and could end up masking real errors.

Reply:
Ok, I will try to add isProducer guards to the call sites

Reply:
We have some code that takes alq as input and is called fro both injectors and producers. For instance https://github.com/OPM/opm-simulators/blob/70152b67ff2d9d9fc9b9136e5c3bb444a51b4000/opm/simulators/wells/WellBhpThpCalculator.cpp#L101

It is possible to refactor those, but maybe the solution of returning 0 with getAlq is called from the well models is sufficient?

Reply:
Also https://github.com/OPM/opm-simulators/blob/70152b67ff2d9d9fc9b9136e5c3bb444a51b4000/opm/simulators/wells/WellBhpThpCalculator.cpp#L226
takes alq as input for both injectors and producers